xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java

```diff
@@ -453,6 +453,86 @@ public void lrsStreamClosedAndRetried() {
     verify(backoffPolicy2, times(1)).nextBackoffNanos();
   }
 
+  @Test
+  public void lrsStreamRetryAndRereport() {
+    verify(mockLoadReportingService).streamLoadStats(lrsResponseObserverCaptor.capture());
+    StreamObserver<LoadStatsResponse> responseObserver = lrsResponseObserverCaptor.getValue();
+    assertThat(lrsRequestObservers).hasSize(1);
+    StreamObserver<LoadStatsRequest> requestObserver = lrsRequestObservers.poll();
+
+    // First LRS request sent.
+    verify(requestObserver).onNext(EXPECTED_INITIAL_REQ);
+    assertThat(logs).containsExactly("DEBUG: Initial LRS request sent: " + EXPECTED_INITIAL_REQ);
+    logs.poll();
+    assertEquals(0, fakeClock.numPendingTasks(LRS_RPC_RETRY_TASK_FILTER));
+
+    // Balancer sends back a normal response.
+    responseObserver.onNext(buildLrsResponse(100));
+    assertThat(logs).containsExactly(
+        "DEBUG: Received LRS initial response: " + buildLrsResponse(100));
+    logs.poll();
+
+    // A load reporting task is scheduled.
+    assertEquals(1, fakeClock.numPendingTasks(LOAD_REPORTING_TASK_FILTER));
+    fakeClock.forwardNanos(99);
+    verifyNoMoreInteractions(requestObserver);
+
+    // Balancer closes the stream with error.
+    responseObserver.onError(Status.UNKNOWN.asException());
+
+    // The unsent load report is cancelled.
+    assertEquals(0, fakeClock.numPendingTasks(LOAD_REPORTING_TASK_FILTER));
+    // Will retry immediately as balancer has responded previously.
+    verify(mockLoadReportingService, times(2)).streamLoadStats(lrsResponseObserverCaptor.capture());
+    assertThat(logs).containsExactly("DEBUG: LRS stream closed, backoff in 0 second(s)",
```

Comment:
You don't need this either. This is just verifying `handleStreamClosed()`.  Stream is closed for sure. `assertThat(lrsRequestObservers).hasSize(1)` is sufficient to verify that backoff is 0 and retry happens.

Reply:
Removed.